packages/lodestar/test/scripts/blsPubkeyBytesFrequency.ts

```diff
@@ -0,0 +1,128 @@
+import fs from "fs";
+import {getClient} from "@chainsafe/lodestar-api";
+import {config} from "@chainsafe/lodestar-config/default";
+import {newZeroedArray} from "@chainsafe/lodestar-beacon-state-transition";
+import SHA256 from "@chainsafe/as-sha256";
+
+// Script to analyze if a raw BLS pubkey bytes are sufficiently even distributed.
+// If so, a shorter slice of the pubkey bytes can be used as key for the pubkey to index map.
+//
+// # How to use
+// ```
+// INFURA_ETH2_URL=https://someurl ../../node_modules/.bin/ts-node test/scripts/blsPubkeyBytesFrequency.ts collisions
+// ```
+// Available commands:
+// - `frequency`
+// - `collisions`
+//
+// # Results
+// - byte pubkey[0] is not evenly distributed, since it includes some flags. Median frequency is 0.
+// - bytes pubkey[1:5] are very evenly distributed.
+//
+// # Collisions rates
+// (not hashed, byte offset = 1)
+// bytes 1, collision rate 1
+// bytes 2, collision rate 0.92230224609375
+// bytes 3, collision rate 0.00013267993927001953
+// bytes 4, collision rate 2.0954757928848267e-9
+//
+// (hashed)
+// bytes 1, collision rate 1
+// bytes 2, collision rate 0.92401123046875
+// bytes 3, collision rate 0.00013625621795654297
+// bytes 4, collision rate 3.026798367500305e-9
```

Comment:
I guess first bytes are extracted and then hashed? (or otherway around?)

Reply:
Because there are 250_000 entries so if you only index by the first byte (256 possible values), 100% of the entries have collisions.